std/assembly/array.ts

```diff
@@ -129,6 +153,20 @@ export class Array<T> {
     return load<T>(this.__memory + <usize>--this.__length * sizeof<T>());
   }
 
+  reduce<U>(
+    callbackfn: (previousValue: U, currentValue: T, currentIndex: i32, array: Array<T>) => U,
+    initialValue: U
+  ): U {
+    var accumulator: U;
```

Comment:
It seems:

```ts
var accumulator: U = initialValue;
```

Reply:
oh, right, i haven't used that at all, thank you :) i'll fix it and add some tests for it

Reply:
done